### PR TITLE
[TD-10418]taosdemo cmdline datatype binary, nchar with customized width

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

**Review by sangshuduo — CHANGES_REQUESTED** [review]
I think 'taosdemo --help' needs to mention this new feature too. 
And we need to add a test case to CI too.

File: src/kit/taosdemo/taosdemo.c [hunk 1]
```diff
@@ -803,7 +804,7 @@ static void printHelp() {
     printf("%s%s%s%s\n", indent, "-q, --query-mode=MODE", "\t\t",
             "Query mode -- 0: SYNC, 1: ASYNC. By default use SYNC.");
     printf("%s%s%s%s\n", indent, "-b, --data-type=DATATYPE", "\t",
-            "The data_type of columns, By default use: FLOAT, INT, FLOAT.");
+            "The data_type of columns, By default use: FLOAT,INT,FLOAT. NCHAR and BINARY can also use custom length. Eg: NCAHR(16),BINARAY(8)");
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c]: NCAHR(16),BINARAY(8)");

typo found.

File: src/kit/taosdemo/taosdemo.c [hunk 2]
```diff
@@ -2735,15 +2738,15 @@ static int printfInsertMeta() {
                     g_Dbs.db[i].superTbls[j].sampleFile);
             printf("      tagsFile:          \033[33m%s\033[0m\n",
                     g_Dbs.db[i].superTbls[j].tagsFile);
-            printf("      columnCount:       \033[33m%d\033[0m\n",
+            printf("      columnCount:       \033[33m%d\033[0m\n        ",
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c:2741]: extra spaces?

File: src/kit/taosdemo/taosdemo.c [hunk 3]
```diff
@@ -2735,15 +2738,15 @@ static int printfInsertMeta() {
...
                     g_Dbs.db[i].superTbls[j].columnCount);
             for (int k = 0; k < g_Dbs.db[i].superTbls[j].columnCount; k++) {
                 //printf("dataType:%s, dataLen:%d\t", g_Dbs.db[i].superTbls[j].columns[k].dataType, g_Dbs.db[i].superTbls[j].columns[k].dataLen);
                 if ((0 == strncasecmp(g_Dbs.db[i].superTbls[j].columns[k].dataType,
                                 "binary", 6))
                         || (0 == strncasecmp(g_Dbs.db[i].superTbls[j].columns[k].dataType,
                                 "nchar", 5))) {
-                    printf("column[\033[33m%d\033[0m]:\033[33m%s(%d)\033[0m ", k,
+                    printf("column[%d]:\033[33m%s(%d)\033[0m ", k,
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c:2749]: why remove colored output?

**Review by sangshuduo — CHANGES_REQUESTED** [review]

File: src/kit/taosdemo/taosdemo.c (hunk 1 above)
Comment (zhaoyanggh) [reply src/kit/taosdemo/taosdemo.c]: I'll fix it

File: src/kit/taosdemo/taosdemo.c (hunk 2 above)
Comment (zhaoyanggh) [reply src/kit/taosdemo/taosdemo.c:2741]: when printing the meta info, I want to keep all the line has the same indent

File: src/kit/taosdemo/taosdemo.c (hunk 3 above)
Comment (zhaoyanggh) [reply src/kit/taosdemo/taosdemo.c:2749]: I saw for the tag we don't have color, so I just keep the column and tag the same.

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**